tutorials/backend/hasura-authentication/tutorial-site/content/integrations/casdoor.md

```diff
@@ -0,0 +1,149 @@
+canonicalUrl: "https://casdoor.org/docs/integration/Haskell/Hasura"
+---
+
+## What is Casdoor?
+
+Casdoor is a UI-first Identity Access Management (IAM) / Single-Sign-On (SSO) platform based on OAuth 2.0, OIDC, SAML
+and CAS.
+
+Casdoor serves both the web UI and the login requests from the application users.
+
+Casdoor comes with features such as:
+
+- Front-end and back-end separate architecture, developed by Golang, Casdoor supports high concurrency, provides
+  web-based managing UI and supports multiple languages(Chinese, English).
+- Casdoor supports third-party applications login, such as GitHub, Google, QQ, WeChat, etc., and supports the extension
+  of third-party login with plugins.
+- With Casbin based authorization management, Casdoor supports ACL, RBAC, ABAC, RESTful accessing control models.
+- Phone verification code, email verification code and password retrieval functions.
+- Accessing logs auditing and recording.
+- Alibaba Cloud, Tencent Cloud, Qiniu Cloud image CDN cloud storage.
+- Customizable registration, login, and password retrieval pages.
+- Casdoor supports integration with existing systems by db sync, so users can transition to Casdoor smoothly.
+- Casdoor supports mainstream databases: MySQL, PostgreSQL, SQL Server, etc., and supports the extension of new
+  databases with plugins.
+
+In this section, you will learn how to integrate Casdoor with Hasura.
+
+## Deploy Casdoor
+
+Firstly, you should deploy a Casdoor instance.
+
+You can refer to the Casdoor official documentation for the
+[Server Installation](https://casdoor.org/docs/basic/server-installation).
+
+After a successful deployment, you need to ensure:
+
+- The Casdoor server is successfully running on **http://localhost:8000**.
+- Open your favorite browser and visit **http://localhost:7001**, you will see the login page of Casdoor.
+- Input `admin` and `123` to test login functionality is working fine.
+
+Then you can quickly implement a Casdoor-based login page in your own app with the following steps.
+
+## Configure Casdoor application
+
+1. Create or use an existing Casdoor application.
+2. Add a redirect url: `http://CASDOOR_HOSTNAME/login`
+   ![Casdoor Application Setting](https://github.com/RanTao123/image/blob/main/Casdoor%20Application%20Setting.png?raw=true)
+3. Copy the client ID, we will need it in the following steps.
+
+## Add user in Casdoor
+
+Now you have the application running, but you must create a user and assign them a role next.
+
+Go to the “Users” page and click “Add user” in the top right corner. This will open a new page where you can add the
+user.
+
+![Pic showing the users page](https://github.com/RanTao123/image/blob/main/user.png?raw=true)
+
+Save the user after adding a username and the organization Hasura (other details are optional).
+
+Now you need to set up a password for your user, which you can do by clicking "Manage Your Password."
+
+Choose a password for your user and confirm it.
```

Comment:
This section is unclear.

Notably, you reference an organization called Hasura, but you never explicitly tell the user to create this organization. The `## Configure Casdoor application` needs to be clearer for a user to follow.